feat: FAQ page (#935)

Stars22 opened the pull request:
close #935
- not sure why button is round in CTA, bug or feature?


![localhost_5173_pages_faq](https://github.com/epicmaxco/vuestic-admin/assets/29259987/4b1056d9-3f97-40bb-a3b8-f23e409077fd)

m0ksem commented:
> not accurate to figma design

We need to update `vuestic-ui` to `1.8.0`, but because it will be released on Monday, you can try to use `0.0.0-experimental-f060346c3-20230830` for now.

asvae commented on src/pages/admin/pages/faq/lib/popularCategories.json:
```diff
@@ -0,0 +1,56 @@
+[
+  {
+    "id": 1,
+    "icon": "diversity_1",
```
That's not the icon in design

Design:
<img width="199" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/dc024065-74bc-4e2e-af5a-3fc390bf954c">

PR:
<img width="177" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/dac820e5-5b0f-4f2f-b5d7-de3515540add">

I see it being available in font. Maybe we use wrong version?
<img width="480" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/39f420a6-2c14-4482-8762-9393d0866d27">

asvae commented on src/pages/admin/pages/faq/FaqPage.vue:
```diff
@@ -3,16 +3,18 @@
     <va-card class="mb-8">
```
There are 2 sources of truth right now:
* design
* vuestic-ui

If there is a conflict between two - we have to discuss (i.e. vuestic-ui is hard to customize or some customization is needed at all).

Vuestic-admin current state is not source of truth and should be ignored completely. (both visuals and structure)


<img width="1194" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/6ed25e38-2116-46ce-b8a2-609a37a1e39c">

asvae commented on src/pages/admin/pages/faq/FaqPage.vue:
```diff
@@ -3,16 +3,18 @@
         </va-input>
       </va-card-content>
     </va-card>
+    <PopularCategories :categories="filteredCategories" />
```
Implementation is off visually

<img width="886" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/b8e95bd5-c49e-43f3-be32-63e833aecafc">

asvae replied:
Still off (less off than before :) )

<img width="719" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/0dfed8d3-231b-4020-ab08-0f6b277a9ba0">

asvae commented on src/pages/admin/pages/faq/FaqPage.vue:
```diff
@@ -3,16 +3,18 @@
 
     <va-card>
-      <va-card-title> Frequently Asked Questions </va-card-title>
+      <va-card-title> <h2>Popular questions</h2> </va-card-title>
```
<img width="409" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/bfa31d7d-6eaa-4fd6-bb3b-d3dcf03547e8">

Please stick to design (communicate you have to wait for collapses etc - you can do so in discord).

asvae replied:
1.8.0 is up by the way, so you can follow up with recent design.

asvae commented on src/pages/admin/pages/faq/FaqPage.vue:
```diff
@@ -3,16 +3,18 @@
     <va-card class="mb-8">
```
I don't see this section implemented at all

<img width="426" alt="image" src="https://github.com/epicmaxco/vuestic-admin/assets/5394573/9fb8f7d8-c6c3-4f26-a33b-d5ed43296a65">

Stars22 replied:
![localhost_5173_pages_faq](https://github.com/epicmaxco/vuestic-admin/assets/29259987/bcbbc4ac-e4d6-44f7-ba6e-62e6dbc859be)
- made visual updates, maybe missed something but tried my best.
- there is still issue with icons(will check them tomorrow)

m0ksem replied:
For sidebar issue, we gonna need to use VaLayout component.

Stars22 replied:
@m0ksem if you speak about sidebar "height" issue it won't affect in browser experience, the sidebar doesn't move on scroll. I just changed(not in the code!) right column overflow value to make full page screenshot.

asvae reviewed (changes requested):
Hey, still needs some work.

asvae commented on src/pages/admin/pages/faq/FaqPage.vue:
```diff
@@ -0,0 +1,92 @@
+          </va-collapse>
+          <va-collapse header="Third question">
+            <div class="p-4 text-sm">
+              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Nemo veniam provident voluptates consequuntur
```
Can we generate some actual text here instead of lorem? You can use chat gpt for that. You can use titles from design.

Stars22 commented:
![localhost_5173_pages_faq](https://github.com/epicmaxco/vuestic-admin/assets/29259987/8b893680-ef4c-4f64-9f24-8d757a8fcada)

Stars22 commented:
![image](https://github.com/epicmaxco/vuestic-admin/assets/29259987/e4794d8e-e33b-4de9-8e83-f9f0f09bfaa5)
updated columns responsivness, updated email modal

asvae commented:
@m0ksem diagonal review from you would be nice here

m0ksem reviewed (changes requested):
Cards shadows look awful.

How about we make shadow color less visible?
```js
  colors: {
    variables: {
      shadow: 'rgba(0, 0, 0, 0.01)'
    }
  }
```

asvae commented:
> Cards shadows look awful.
> 
> How about we make shadow color less visible?
> 
> ```js
>   colors: {
>     variables: {
>       shadow: 'rgba(0, 0, 0, 0.01)'
>     }
>   }
> ```

Moved fix in ui here - https://github.com/epicmaxco/vuestic-ui/issues/3964

+ Fixed in admin
